object.c

```diff
@@ -520,7 +520,9 @@ rb_obj_size(VALUE self, VALUE args, VALUE obj)
  *
  *  Yields <i>obj</i> and returns the result.
  *
- *	'my string'.yield_self {|s|s.upcase} #=> "MY STRING"
+ *  require "ostruct"
+ *  user = OpenStruct.new(firstname: "John", lastname: "Doe")
+ *  user.yield_self { |u| "#{u.firstname} #{u.lastname}" } #=> "John Doe" 
```

Comment:
feel free to take the examples from the my article if they convince you https://bogdanvlviv.github.io/posts/ruby/new-method-kernel-yield_self-since-ruby-2_5_0.html